ui/nuxeo-data-table/iron-data-table.js

```diff
@@ -224,7 +225,11 @@ import '../nuxeo-button-styles.js';
 
           <div id="header">
             <nuxeo-data-table-row header>
-              <nuxeo-data-table-checkbox header hidden$="[[!selectionEnabled]]"></nuxeo-data-table-checkbox>
+              <nuxeo-data-table-checkbox
+                header$="[[!_isSelectionEnabled(selectionEnabled, selectAllEnabled)]]"
```

Comment:
I get your point, but notice that I didn't change the current behavior. If you check [here](https://github.com/nuxeo/nuxeo-elements/blob/aacc16abf274c18518583090f92ff376bcd5f9cc/ui/nuxeo-data-table/data-table-checkbox.js#L31), we were already doing it. Furthermore, if selection is enabled but not select all, we cannot hide the checkbox, but just set its visibility attribute, otherwise the row will collapse to the left. 

That said, this is clearly poor design. Let me review and see what I can do to improve it.

Reply:
@Gabez0r do you have any idea what we could here to improve? 
My main problem is that we need to something like this:

`
display: block; 
visibility: hidden;
`

Reply:
The `header` attribute in the `nuxeo-data-table-checkbox` is a weird one, as we toggle it on and off even when already inside a header. 🤔  What about computing the class of the `nuxeo-data-table-checkbox` with a computed binding (e.g. `nuxeo-data-table-checkbox class$="_compute..."`)? Would this be better?

Reply:
What do you think about the solution I used? It is similar to the we are already using the table.

Reply:
I can live with it, but this should be a target for further improvement.

Reply:
It looks much better. I would favor using classes instead of styles, so we separate presentation from styling, but if it is not possible here, I'm ok with it.

Reply:
I was trying with classes because it makes the code cleaner and allows us to separate between presentation and styling as you said, but it is not working... I'll ping you to try it with me.